packages/core/core/src/requests/BundleGraphRequest.js

```diff
@@ -90,24 +89,70 @@ export default function createBundleGraphRequest(
 ): BundleGraphRequest {
   return {
     type: 'bundle_graph_request',
-    id: 'BundleGraph:' + input.assetGraph.getHash(),
+    id: 'BundleGraph',
     run: async input => {
+      let {options, api, invalidateReason} = input;
+      let {optionsRef, requestedAssetIds, signal} = input.input;
+      let request = createAssetGraphRequest({
+        name: 'Main',
+        entries: options.entries,
+        optionsRef,
+        shouldBuildLazily: options.shouldBuildLazily,
+        requestedAssetIds,
+      });
+      let {assetGraph, changedAssets, assetRequests} = await api.runRequest(
+        request,
+        {
+          force: options.shouldBuildLazily && requestedAssetIds.size > 0,
+        },
+      );
+
+      assertSignalNotAborted(signal);
+
+      // If any subrequests are invalid (e.g. dev dep requests or config requests),
+      // bail on incremental bundling. We also need to invalidate for option changes,
+      // which are hoisted to direct invalidations on the bundle graph request.
+      let subRequestsInvalid =
```

Comment:
Does this mean that changes like adding a runtime would stop incremental bundling from happening? Thats the only issue I encountered when trying to not apply runtimes for incremental.

Reply:
Yeah, if any config changes or plugin changes occurred, then bundling will not be incremental. I think this should be pretty rare so probably acceptable?